src/platform/BL_Python/platform/feature_flag/feature_flag_router.py

```diff
@@ -9,60 +7,43 @@ class FeatureFlagRouter(ABC):
     All feature flag routers should extend this class.
     """
 
-    _logger: Logger
-    _feature_flags: Dict[str, bool]
-
-    def __init__(self, logger: Logger) -> None:
-        self._logger = logger
-        self._feature_flags = {}
-        super().__init__()
-
-    def set_feature_is_enabled(self, name: str, is_enabled: bool) -> None:
+    @abstractmethod
+    def _notify_enabled(self, name: str) -> None:
         """
-        Enables or disables a feature flag in the in-memory dictionary of feature flags.
-
-        Subclasses should call this method to validate parameters and cache values.
+        Override to provide a method to be used to notify when a feature is enabled.
+        Implementation of when and whether this is called is the responsibility of subclasses.
+        This is never called by default.
 
-        name: The feature flag to check.
+        :param str name: The name of the feature flag.
+        """
 
-        is_enabled: Whether the feature flag is to be enabled or disabled.
+    @abstractmethod
+    def _notify_disabled(self, name: str) -> None:
         """
-        if name in self._feature_flags:
-            self._logger.warn(
-                f"Overridding feature flag value for '{name}'. Toggling from {self._feature_flags[name]} to {self._feature_flags[name]}"
-            )
-        if type(name) != str:
-            raise TypeError("`name` must be a string.")
+        Override to provide a method to be used to notify when a feature is disabled.
+        Implementation of when and whether this is called is the responsibility of subclasses.
+        This is never called by default.
```

Comment:
Thanks for catching this. Python and pyright so helpfully allow subclasses not to override abstract methods _until the class is instantiated_ in which case it will crash the application with `TypeError: Can't instantiate abstract class ... with abstract method ...`.

Since this base class is intended only to enforce the get/set for the flag, I will make these methods NOOPs. If the `notify` methods aren't overridden, that's really the decision of the subclass and I do not want the base class to decide that that should be an error.

Reply:
Armchair speculating, I think it would be less surprising for the user if the methods raised `NotImplementedError` rather than just being no-ops, as then there would be no ambiguity about what the base class was providing... but that's not a really important detail and is subject to debate.

Reply:
It's a fair point. I'm about to push some changes around this - maybe it addresses this?

Reply:
To recap on conversation - as long as the NOOP and implementation responsibility are clearly documented, we can NOOP all day.